src/main/java/com/ncedu/cheetahtest/service/mail/HtmlMailImpl.java

```diff
@@ -29,4 +31,33 @@
 
     return Optional.empty();
   }
+
+  @Override
+  public Optional<String> getHtmlWithStrings(List<String> messages, String htmlPath) {
+
+    File file = new File(htmlPath);
+
+    StringBuilder htmlStringBuilder = new StringBuilder();
+
+    try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+
+
+      String tmp = "";
+      int i = 0;
+      while (reader.ready()) {
+        tmp = reader.readLine();
+        while (tmp.contains("%s")){
+          tmp = tmp.replace("%s", messages.get(i++));
+        }
+        htmlStringBuilder.append(tmp);
+      }
+
+      return Optional.of(htmlStringBuilder.toString());
+
+    } catch (IOException e) {
+      e.getMessage();
+    }
+
+    return Optional.empty();
+  }
```

Comment:
Please, use String.format for this purpose. Moreover. you can change signature of getHtmlWithStringInside with several messages and overload it with 1 message parameter. 
You could also take a look to template engines for emails instead of custom implementation.